src/plugins/ucc_discovery_plugin.cpp

```diff
@@ -0,0 +1,345 @@
+          // predicate may be swapped, so get proper operand
+          auto column_candidate = std::static_pointer_cast<LQPColumnExpression>(join_predicate->left_operand());
+          if (!expression_evaluable_on_lqp(column_candidate, *subtree_root.get())) {
+            column_candidate = std::static_pointer_cast<LQPColumnExpression>(join_predicate->right_operand());
+          }
+
+          auto candidates = generate_valid_candidates(subtree_root, column_candidate);
+          if (candidates) {
+            for (const auto& candidate : *candidates) {
+              ucc_candidates.insert(candidate);
+            }
+          }
+          break;
+        }
+
+        case JoinMode::Inner: {
+          auto column_candidates = std::vector<std::shared_ptr<AbstractExpression>>{join_predicate->left_operand(),
+                                                                                    join_predicate->right_operand()};
+          for (const auto& column_candidate : column_candidates) {
+            const auto casted_candidate = std::static_pointer_cast<LQPColumnExpression>(column_candidate);
+
+            // determine which subtree (left or right) belongs to the ColumnExpression
+            auto subtree_root = join_node.left_input();
+            if (!expression_evaluable_on_lqp(column_candidate, *subtree_root.get())) {
+              subtree_root = join_node.right_input();
+            }
+
+            // for Join2Semi, this column is already interesting for optimization, add it as a potential candidate
+            const auto table = std::static_pointer_cast<const StoredTableNode>(casted_candidate->original_node.lock());
+            ucc_candidates.insert(UCCCandidate{table->table_name, casted_candidate->original_column_id});
+
+            auto candidates = generate_valid_candidates(subtree_root, casted_candidate);
+            if (candidates) {
+              for (const auto& candidate : *candidates) {
+                ucc_candidates.insert(candidate);
+              }
+            }
+          }
+          break;
+        }
+
+        case JoinMode::Semi:
+        case JoinMode::Left: {
+          // want to check only the right hand side here, as this is the one that will be removed in the end
+          auto subtree_root = join_node.right_input();
+          // predicate may be swapped, so get proper operand
+          auto column_candidate = std::static_pointer_cast<LQPColumnExpression>(join_predicate->right_operand());
+          if (!expression_evaluable_on_lqp(column_candidate, *subtree_root.get())) {
+            column_candidate = std::static_pointer_cast<LQPColumnExpression>(join_predicate->left_operand());
+          }
+
+          auto candidates = generate_valid_candidates(subtree_root, column_candidate);
+          if (candidates) {
+            for (const auto& candidate : *candidates) {
+              ucc_candidates.insert(candidate);
+            }
+          }
+          break;
+        }
+
+        default: {
```

Comment:
Not really needed, as this is what happens in line 335 anyways, without any more code being executed in between.
I do see though, that it may add some layer of safety, if someone were to write code in between the end of the switch case and the return in the future.

Is the (currently unnecessary) code duplication worth that security in the future in your eyes?

Reply:
I think it's better not to leave a case empty. If I'm not missing something on a quick look, you can also think of replacing your `break`s by `return`s and be a little more explicit there (so one directly sees what happens next without scrolling).